MiniApp/Classes/core/Storage/MiniAppKeyChain.swift

```diff
@@ -74,11 +76,78 @@ import Foundation
         return supportedPermissionList
     }
 
-    private func write(keys: KeysDictionary) {
+    private func writePermissionInfo(keys: KeysDictionary) {
         guard let data = try? JSONEncoder().encode(keys) else {
             return
         }
+        setValueInKeyChain(data: data)
+    }
+
+    private func retrieveAllPermissions() -> KeysDictionary? {
```

Comment:
It seems like this `MiniAppKeyChain` has too many purposes now. It's handling interactions with the KeyChain, parsing of permissions, and parsing of verification data. I think it should have just a single purpose: interacting with the KeyChain.

So maybe we should have three separate classes: `MiniAppKeyChain`, `MiniAppPermissionsStorage`, and `MiniAppVerificationStorage`. But probably that could be changed in another PR instead of this one.

What do you think?

Reply:
That's a good idea. Done 👍